### PR TITLE
[BEAM-1689] Apply changes for Flink's StatefulDoFnRunner to the primary StatefulDoFnRunner

### PR DESCRIPTION
**Pull request by aljoscha**
R: @kennknowles

### CODE REVIEW

**Comment by aljoscha** [comment]
Run Flink RunnableOnService

**Comment by aljoscha** [comment]
Run Flink RunnableOnService

File: runners/core-java/src/test/java/org/apache/beam/runners/core/StatefulDoFnRunnerTest.java [hunk 1]
```diff
@@ -190,7 +186,7 @@ public void testGarbageCollect() throws Exception {
 
     // advance watermark past end of WINDOW_2 + allowed lateness
     advanceInputWatermark(
-        timerInternals, WINDOW_2.maxTimestamp().plus(ALLOWED_LATENESS + 1), runner);
+        timerInternals, WINDOW_2.maxTimestamp().plus(ALLOWED_LATENESS + 2), runner);
```
Comment (kennknowles) [inline runners/core-java/src/test/java/org/apache/beam/runners/core/StatefulDoFnRunnerTest.java]: You might want to make the `2` self-documenting by something like `TimerInternalsCleanupTimer.GC_DELAY + 1` or something. Not a big deal.

Comment (aljoscha) [inline runners/core-java/src/test/java/org/apache/beam/runners/core/StatefulDoFnRunnerTest.java]: will do!

**Pull request closed without merging**